src/main/groovy/org/savantbuild/plugin/linter/LinterPlugin.groovy

```diff
@@ -54,9 +57,28 @@ class LinterPlugin extends BaseGroovyPlugin {
     boolean reportSuppressedViolations = attributes.getOrDefault("reportSuppressedViolations", false)
     String languageVersion = attributes.getOrDefault("languageVersion", "17")
     String inputPath = attributes.getOrDefault("inputPath", "src/main/java")
+    String sourceClassesPath = attributes.getOrDefault("sourceClassesPath", "build/classes/main")
+    String testClassesPath = attributes.getOrDefault("testClassesPath", "build/classes/test")
+    String cachePath = attributes.getOrDefault("cachePath", "build/pmd.cache")
     String minimumPriority = attributes.getOrDefault("minimumPriority", "MEDIUM")
     String[] ruleSets = attributes.getOrDefault("ruleSets", [])
     boolean failOnViolations = attributes.getOrDefault("failOnViolations", true)
+    String customRuleDependencyGroup = attributes.getOrDefault("customRuleDependencyGroup", null)
+    def ruleClassLoader = RuleSetLoader.class.getClassLoader()
+    if (customRuleDependencyGroup) {
+      def rules = new DependencyService.TraversalRules().with(customRuleDependencyGroup,
+          new DependencyService.TraversalRules.GroupTraversalRule(false, false))
+      def graph = project.dependencyService.resolve(project.artifactGraph, project.workflow, rules)
+
+      URL[] ruleURLs = graph.toClasspath()
+          .paths
+          .collect { p -> p.toUri().toURL() }.toArray(new URL[0])
+      ruleClassLoader = new URLClassLoader(ruleURLs,
+          ruleClassLoader)
+
+      output.infoln("Including the following custom rules in the PMD classpath: [%s]",
+          ruleURLs)
+    }
```

Comment:
This seems sort of dangerous - maybe ok since it is just a build time dep thing?

Reply:
I don't see any danger here.

Reply:
To be clear, what I mean is, if you allow the user to add jars to the class loader - are we correctly evaluating risk?

Reply:
This is sorta what I was talking about in the above code. I'm not sure how this ClassLoader will be used, but it might run into issues based on the hierarchy of ClassLoaders from the primordial one that starts the JVM, to the Plugin one, to the one here.